src/GC/GC.jl

```diff
@@ -3,77 +3,134 @@
 
 Garbage collection of Python objects.
 
-See `disable` and `enable`.
+See [`gc`](@ref).
 """
 module GC
 
 using ..C: C
 
-const ENABLED = Ref(true)
-const QUEUE = C.PyPtr[]
+const QUEUE = (; items = C.PyPtr[], lock = Threads.SpinLock())
```

Comment:
Base has `Lockable` starting in 1.11 (exported in 1.12) FYI. So not useful here but just so you know :). It is just like this, except it also enforces the lock is held when you access the item it holds via indexing.